### DISCUSSION TITLE
Identifying connected components across tiled images

### DISCUSSION THREAD

**Discussion by jchubber** [seung-lab/connected-components-3d#97] [Q&A] [answered]
Is anyone aware of methods for identifying connecting components that span across tiled images _when the are images are too large to be combined before CC3D analysis_? I am searching through procedurally generated data, which has a very large (assume infinite) space. I can use CC3D to analyze arrays of roughly 5000x5000x128 before I start to get low on RAM. Unfortunately, the components I'm measuring can sometimes be much larger than that (I'm actually not sure exactly how large these components get, which is the purpose of my research).

Ideas I've considered so far:
1. Get access to machines with more RAM
2. Analyze multiple 5000x5000x128 images and then write some new custom code that looks at the boundary layers where images meet and creates an equivalency table just for the boundaries (and then stitch together the components found across each image using the boundary equivalency table.
3. Decrease the image resolution greatly, for example by downsampling the data, so that I can run CC3D on a larger span of the data (i.e., starting with 20000x20000x128 but dropping enough data to downsample that to the 5000x5000x128 that I can run with my current RAM). The problem here is not being able to know for sure if the component is **_truly_** connected at the highest resolution.

I would welcome any and all questions, ideas, or advice. And thank you again for this great library!

**Comment by william-silversmith** [comment, marked as answer]
Hi Mr. Chung, 

Yes, in fact I do have an scalable version of connected components here: 

https://github.com/seung-lab/igneous#connected-components-labeling-ccl-beta

However, it is designed for datasets that are in Neuroglancer Precomputed format. Why don't you take a look and let me know if you have questions? 

Here are some related repos that might help you understand what you're looking at:
- https://github.com/google/neuroglancer
- https://github.com/seung-lab/cloud-volume
